tests/core/pyspec/eth2spec/test/altair/block_processing/test_process_sync_committee.py

```diff
@@ -238,6 +223,17 @@ def test_sync_committee_rewards_duplicate_committee(spec, state):
     yield from run_successful_sync_committee_test(spec, state, committee, committee_bits)
 
 
+@with_all_phases_except([PHASE0, PHASE1])
+@spec_state_test
+@always_bls
+def test_sync_committee_rewards_not_full_participants(spec, state):
+    committee = spec.get_sync_committee_indices(state, spec.get_current_epoch(state))
+    rng = random.Random(1010)
+    committee_bits = [rng.choice([True, False]) for _ in committee]
+
+    yield from run_successful_sync_committee_test(spec, state, committee, committee_bits)
+
+
```

Comment:
yep, this should avoid hitting the infinite loop condition :)

are you worried about the (small) chance that `choice` returns `True` for all members? could toss an `assert` in there as a stop-gap...

Reply:
nah, the rng is fixed so it's deterministic now

we *should* make sure we test it with empty participant list though